#165546651 refactored documentation

emmsdan opened the pull request:
#### What does this PR do?
- refactored documentation
- all routes are represented and properly documented with their error responses

#### How should this be manually tested?
- clone repo
- run `npm install`
- run `npm run start:dev`
- open: `http://localhost:3000/documentation`

#### Any background context you want to provide?
- initially, the documentation was not properly written and most of the routes were missing.

#### What are the relevant pivotal tracker stories? (if applicable)
#165546651

#### Screenshots
<img width="1440" alt="Screenshot 2019-04-24 at 5 24 37 PM" src="https://user-images.githubusercontent.com/3631446/56676480-4c32f600-66b6-11e9-8bc6-220b0d595c95.png">

Commit:
refactored documentation

emmsdan committed:
Merge branch 'develop' into bg-refactor-documentation-165546651

igbominadeveloper commented on src/routes/auth.route.js:
```diff
@@ -11,85 +11,16 @@ import {
 
 const authRouter = Router();
```
Thanks for coming through for us man. 
So all the route documentation were actually destined to eventually `die`? because you have removed everything now. I thought they would always be here.

emmsdan replied:
Oh, yeah, I had to **move** it not **remove**. The documentation in those routes files were mostly duplicates and not properly documented. 
So to avoid duplication, I created a new file called: `documentation.route.js` that stores all the documentation for the routes.
I also thought it would always be here, but moving it was important for us to keep the documentation and the routes files clean.

Japhethca commented:
I see that you have removed a couple of documentation for some routes, please can you provide a description on why the docs were removed and how the docs works now.

Japhethca commented on src/routes/controllers/users.controller.js:
```diff
@@ -93,6 +93,7 @@ export async function getAuthorProfile(req, res) {
 export const updateUserProfile = async (req, res) => {
   const { id } = req.user;
   const { firstName, lastName, username, bio, image } = req.body;
+  const user = await getUserbyId(id);
```
Why was this line added?

emmsdan replied:
I used it to test the documentation. forgot to remove it.

emmsdan commented:
> I see that you have removed a couple of documentation for some routes, please can you provide a description on why the docs were removed and how the docs works now.

I moved the documentation to `documentation.route.js` because of most the doc. where duplicated and reused, and also did not contain proper `request and response body`, to fix that, I moved them into a separate file and refactored the documentation that will handle them.
